watcher/diffWatcher.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package watcher
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/sergi/go-diff/diffmatchpatch"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+// DiffWatcher regularly check data at path provided, notifies if changes detected.
+type DiffWatcher struct {
+	latest []byte
+
+	synced []byte
+	diffs  []diffmatchpatch.Diff
+	mu     sync.Mutex
```

Comment:
Maybe add a comment to state what part of the struct is protected by this mutex?

Reply:
Done.